src/components/input/input.js

```diff
@@ -777,14 +777,6 @@ function ngMessagesAnimation($q, $animateCss) {
 function ngMessageAnimation($animateCss) {
   return {
     enter: function(element, done) {
-      var messages = getMessagesElement(element);
-
-      // If we have the md-auto-hide class, the md-input-invalid animation will fire, so we can skip
-      if (messages.hasClass('md-auto-hide')) {
-        done();
-        return;
```

Comment:
But in that case consecutive errors weren't being animated in. E.g. when the input is empty, the `required` error is shown, but if I enter an invalid date afterwards, it's error message stays at `opacity: 0` and `margin: -100%`. The only workaround that I found were removing that line or using `md-auto-hide`.